### DISCUSSION TITLE
ipv6 IP only server support

### DISCUSSION THREAD

**Discussion by mathmath5151** [pivpn/pivpn#1574] [Support]
Hi,
My ISP is only providing ipv6, so if I want to access my Raspberry from outside my only choice is to do it by accessing its ipv6 address.
Unfortunately, after having installed the latest version (master as of 07/24/22) of PIVPN with Wireguard, the handshake is always failing when I try to connect a device to the server.
It worked perfectly when my ISP provided ipv4. I know the Raspberry is accessible from the outside as I have a Tomcat running and can connect to pages I host from my cell phone (not connected to my network).
Am I missing something my use case is just no supported ?

Thanks for your help

**Comment by mathmath5151** [comment]
I managed to make it work by changing the endpoint in the client config. Instead of putting my dynamic DNS address, I put directly the IPv6 address of my server.
The DNS is configured with a AAAA record pointing to that same IPv6 address so I assume it should work (like I mentioned in the OP, reaching my Tomcat default page works using the same DNS).
Maybe it's a Wireguard bug ?

**Reply by TinCanTech** [reply]
Do you have a log file or other output from Wireguard ?

**Reply by cubezucker** [reply]
I understand you managed to resolve the problem and your wireguard connection works?

I'm in the same situation with only a public IPv6 address by my ISP.
Would you care to elaborate how you got your config working?
